### PR TITLE
treewide: standardize gitPackage

### PR DESCRIPTION
**Pull request by khaneliman**
Follow up to https://github.com/nix-community/nixvim/pull/2112

### CODE REVIEW

File: plugins/git/neogit/default.nix [hunk 1]
```diff
@@ -102,6 +101,18 @@ helpers.neovim-plugin.mkNeovimPlugin config {
     };
   };
 
+  extraOptions = {
+    gitPackage = helpers.mkPackageOption {
+      name = "git";
+      default = pkgs.git;
+    };
+
+    whichPackage = helpers.mkPackageOption {
```
Comment (khaneliman) [inline plugins/git/neogit/default.nix:110]: should I move this out into a new category PR?

**Comment by traxys** [comment]
Weren't commitia & fugitive forgotten in this?

**Comment by khaneliman** [comment]
> Weren't commitia & fugitive forgotten in this?

Not sure how I missed em, i'll update.

File: plugins/git/neogit/default.nix [hunk 2]
```diff
@@ -122,8 +128,9 @@ helpers.neovim-plugin.mkNeovimPlugin config {
           "fzf-lua"
         ];
 
-    extraPackages = optional (hasInfix "which" (
-      cfg.settings.commit_view.verify_commit.__raw or ""
-    )) pkgs.which;
+    extraPackages = [
+      cfg.gitPackage
+    ] ++ optional (hasInfix "which" (cfg.settings.commit_view.verify_commit.__raw or "")) pkgs.which;
```
Comment (MattSturgeon) [inline plugins/git/neogit/default.nix:133]: I wonder if for neogit we need a more freeform approach, e.g. `attrsOf (nullOr package)`.

Or at least a `whichPackage` option.

Saying that, is it even possible to do anything without the `which` command? Installing the dependency seems a little overkill. Maybe it's for cross-platform reasons 🤷

**Review by MattSturgeon — APPROVED** [review]
Seems fine at first glance 😁

File: plugins/git/neogit/default.nix (hunk 2 above)
Comment (khaneliman) [reply plugins/git/neogit/default.nix:133]: I have a whichPackage in https://github.com/nix-community/nixvim/pull/2114

**Comment by khaneliman** [comment]
@mergifyio queue

**Pull request merged**